test/index.js

```diff
@@ -82,3 +82,44 @@ test('inheritance', t => {
   t.equal(withoutNew.test2(), expected, 'withoutNew inherited proto method call')
   t.end()
 })
+
+test('factory inheritance', t => {
+  const expected = Symbol()
+  class A {
+    constructor(name) {
+      this.name = name
+    }
+    test() {
+      return expected
+    }
+  }
+  A.property = expected
+
+  const B = f(A)
+
+  class C extends B {
+    constructor(name) {
+      super(name.toUpperCase())
+    }
+    test2() {
+      return super.test()
+    }
+  }
+
+  const D = f(C)
+
+  const withNew = new D('withNew')
+  const withoutNew = D('withoutNew')
+
+  t.equal(withNew.constructor, C, 'withNew has the correct constructor')
+  t.equal(withoutNew.constructor, C, 'withoutNew has the correct constructor')
```

Comment:
It's debatable whether or not these should have `constructor == C` or `constructor == D`.